### DISCUSSION TITLE
I can't configure vertical claiming

### DISCUSSION THREAD

**Discussion by adriangalanuser** [GriefPrevention/GriefPrevention#1585] [Q&A]
Hello!
I've been struggling to configure the plugin to limit my players claiming territory vertically. By default their claims go all the way to bedrock and all the way to the clouds.

So I've checked the documentation and noticed there is a "Claims.MaximumDepth" variable on the config that handles this. I went to the config file and changed it's value to 60, which is about the standard "ground" height on my world. I expected the claims to reach only from 60 in the Y coordinate, but then I went all the way down to bedrock and still the claimed territory was claimed all the way down to the end of the world.

I've tried changing this variable many times, tried high numbers, low numbers and nothing seems to work.

**Comment by bigpresh** [comment]
When you say the territory was claimed all the way down, do you mean that it shows as claimed when you right-click with a stick, or that you cannot place/break blocks there?  The former would be expected I believe, the latter not so much.

> So I've checked the documentation and noticed there is a "Claims.MaximumDepth" variable on the config that handles this.

~~Hmm - it would appear that that setting, whilst documented, doesn't *actually* do anything as far as I can see - it's read from the config and assigned to `this.config_claims_maxDepth` but never referenced by anything else:
https://github.com/TechFortress/GriefPrevention/search?q=this.config_claims_maxDepth~~

EDIT: the above was a load of crap - I copied & pasted too much into the search, we [refer to it elsewhere in as `GriefPrevention.instance.config_claims_maxDepth`](https://github.com/TechFortress/GriefPrevention/search?q=config_claims_maxDepth)

Have you read the documentation for the `Claims.ExtendIntoGroundDistance` setting under [Land Claim Settings in Setup and Configuration](https://github.com/TechFortress/GriefPrevention/wiki/Setup-and-Configuration#land-claim-settings) on the wiki?  It explains how claims will, by default, only extend into the ground a certain distance when first created (so that people can still mine underneath) but automatically grow downwards when the claim owner places/breaks blocks below the claim - so that if the claim owner mines down, their mines are then protected by their claim.

EDIT: that should of course still be limited to the max depth you'd set.

Already-existing deep claims would still continue to be protected, the max depth setting would just stop new claims being created too deep and/or extended below that depth.

It would probably be beneficial to raise an issue instead, filling out the issue template - in particular supplying the config and versions in use.
Can you share your config to make sure it's set appropriatelY

**Reply by adriangalanuser** [reply]
1. I mean both of them. I used a stick and it says it's claimed by a player, and if I try to remove a block, it won't let me.
2. I took a look into that setting as well, played a bit with it, nothing changed. By default it was set to 5, so it doesn't make much sense to me to have that parameter allowing players to reach bedrock :(
3. I thought that as well, so I tried by removing claims and asking my players to claim their territories again. The problem still persists.
4. How do I check the plugin version?

**Reply by bigpresh** [reply]
> I took a look into that setting as well, played a bit with it, nothing changed. By default it was set to 5, so it doesn't make much sense to me to have that parameter allowing players to reach bedrock :(

The parameter determines how many blocks the claim extends below the point you create the claim.  So, e.g. if you create a claim whose corners are both at y level 65 and `Claims.ExtendIntoGroundDistance` is set to 5, then the claim will protect all blocks down to level 60 (but will extend further if the player places/breaks blocks lower)

> How do I check the plugin version?

It will be logged to the server log at startup.

You could consider [downloading a CI build](https://ci.appveyor.com/project/RoboMWM39862/griefprevention/build/artifacts) usual "test builds, there may be problems" warning applies.

**Reply by adriangalanuser** [reply]
So I've clicked your link to download that build, and selected the artifact. I uploaded it to the server replacing the version I had and now the plugin won't load at all.

Console message:
```
15.09 22:27:44 [Server] ERROR Could not load 'plugins/GriefPrevention.jar' in folder 'plugins'
15.09 22:27:44 [Server] INFO org.bukkit.plugin.InvalidPluginException: Unsupported API version 1.17
```

**Comment by Jikoo** [comment]
Note that if you're running 16.17.1 and not a dev build, you're probably running into an issue that was fixed by #1260.

**Reply by adriangalanuser** [reply]
How can I know my plugin version? I downloaded it a few days ago and can't remember

**Reply by Jikoo** [reply]
`/version <name of plugin>`

**Reply by adriangalanuser** [reply]
Yeah, I have the latest one

**Reply by Jikoo** [reply]
Could you please paste the output of the command? Latest is never a valid version, that's the easiest way to cause a misunderstanding in reproduction.

**Reply by adriangalanuser** [reply]
Oooooh sorry I got you wrong. Yeah, I have the 16.17.1 version, I thought you meant the bug was solved in that version.
So how do I get a "dev build"?

**Reply by Jikoo** [reply]
BigPresh did link it in his reply, but the CI is [here](https://ci.appveyor.com/project/RoboMWM39862/griefprevention/build/artifacts). Output of `/version GriefPrevention` should be something like 16.17.1-e6ce386 for the current latest build.

**Reply by adriangalanuser** [reply]
I downloaded that one, but I get this error on server load:
```
15.09 22:27:44 [Server] ERROR Could not load 'plugins/GriefPrevention.jar' in folder 'plugins'
15.09 22:27:44 [Server] INFO org.bukkit.plugin.InvalidPluginException: Unsupported API version 1.17
```

**Reply by Jikoo** [reply]
Yes, the current build requires 1.17. You can go back in the history to try and obtain a build that supports 1.16, but dev builds are only kept for 1 month by the CI.
